.clang-tidy

```diff
@@ -1,10 +1,24 @@
-Checks: >
-  bugprone-*,
-  performance-*,
-  modernize-*,
-  readability-*,
-  cppcoreguidelines-*
-  -modernize-use-trailing-return-type
+Checks: "*,
+        -abseil-*,
+        -altera-*,
+        -android-*,
+        -fuchsia-*,
+        -google-*,
+        -llvm*,
+        -modernize-use-trailing-return-type,
+        -zircon-*,
+        -readability-else-after-return,
+        -readability-static-accessed-through-instance,
+        -readability-avoid-const-params-in-decls,
+        -cppcoreguidelines-non-private-member-variables-in-classes,
+        -misc-non-private-member-variables-in-classes,
+        -include-what-you-use,
```

Comment:
why disable include what you use?

Reply:
You got me excited that there is such a check.  I don't see it: https://clang.llvm.org/extra/clang-tidy/checks/list.html
Though they did add some header checks which will be useful for us.  But I was really hoping for one that'll tell us when we can get away with a forward declaration, which I'm not seeing.

Reply:
I assume this list isn't vetted for our repo.  Let's add a comment so when we have time to scan our repo and find the proper ignore list we'll know to do so and not trust this list here.